### PR TITLE
docs: add middleware guide

### PR DESCRIPTION
**Pull request by ldayananda**

### CODE REVIEW

File: docs/guides/middleware.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Table of Contents
+
+* [Understanding Middleware](#understanding-middleware)
+* [Using Middleware](#using-middleware)
+* [setSource](#setsource)
+
+## Understanding Middleware
+
+Middleware are functions that return an object with methods matching those on the `Tech`. There are currently a limited set of allowed methods that will be understood by middleware, and can be referenced in the [middleware source](https://github.com/videojs/video.js/blob/master/src/js/tech/middleware.js).
```
Comment (gkatsev) [inline docs/guides/middleware.md]: I think we should just list the allowed ones here.

File: docs/guides/middleware.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```
++----------+                      +----------+
+|          |  setter middleware   |          |
+|          +---------------------->          |
+|  Player  |                      |   Tech   |
+|          <----------------------+          |
+|          |  getter middleware   |          |
++----------+                      +----------+
+```
```
Comment (gkatsev) [inline docs/guides/middleware.md]: nice

File: docs/guides/middleware.md (hunk 1 above)
Comment (ldayananda) [reply docs/guides/middleware.md]: I suppose the allowed methods won't be updated that frequently that it becomes difficult to maintain 🤔 Sounds good

Comment (gkatsev) [reply docs/guides/middleware.md]: yeah, I'd hope that the next time this is updated, we just open it fully.

Comment (ldayananda) [reply docs/guides/middleware.md]: Updated!

**Comment by gkatsev** [comment]
Nice! https://deploy-preview-4877--videojs-docs.netlify.com/tutorial-middleware.html

**Pull request merged**